app/services/moment_keywords.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+class MomentKeywords
+  def initialize(moment)
+    @moment = moment
+    @moment_keywords = []
+  end
+
+  def extract_moment_keywords
+    extract_keywords(@moment.categories)
+    extract_keywords(@moment.moods)
+    extract_keywords(@moment.strategies)
+    @moment_keywords.push(extract_name, extract_why, extract_fix)
+    remove_special_chars
+    downcase_keywords
+  end
+
+  private
+
+  def strip_html(str)
+    ActionController::Base.helpers.strip_tags(str)
+  end
+
+  def extract_keywords(array)
+    array.each do |item|
+      @moment_keywords.push(item['name'].split,
+                            strip_html(item['description']).split)
+    end
+  end
+
+  def extract_name
+    @moment.name.split
+  end
+
+  def extract_why
+    strip_html(@moment.why).split
+  end
+
+  def extract_fix
+    strip_html(@moment.fix).split
+  end
```

Comment:
The shape of these methods are identical. Could be a good opportunity to DRY this up a bit! Perhaps a general `extract` method that takes in an argument? This new method [could also be used above for this line](https://github.com/ifmeorg/ifme/blob/78adeaa44bc1dd06bd969b0fa7561df3da67b25f/app/services/moment_keywords.rb#L27).